Arm backend: Add 16A8W support for view and transpose operations

Ninja91 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13802
* #13801
* #13800
* __->__ #13799
* #13798
* #13797
* #13796

Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218

Ninja91 committed:
Add 16A8W support for view and transpose operations

Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

[ghstack-poisoned]

digantdesai reviewed (approved):
try to fix the code dup issue

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support for view and transpose operations"


Add 16A8W quantization support for view and transpose operations in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, tanh, and slice operations, extending int16 support to view and transpose operations.

Changes:
- Add INT16 dtype validation support in op_transpose.py
- Add test_view_tensor_16a8w_tosa_INT test function
- Enable test_view.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511313](https://our.internmc.facebook.com/intern/diff/D80511313/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]
